src/bin/pg_autoctl/fsm_transition.c

```diff
@@ -917,7 +926,7 @@ fsm_promote_standby(Keeper *keeper)
 		return false;
 	}
 
-	if (!primary_disable_synchronous_replication(postgres))
+	if (!fsm_disable_replication(keeper))
```

Comment:
Mmm, it seems ok to share code that way for now, yes.

Reply:
I guess the main reason it was not like that before was because the standby would not have a replication slot, so there would be nothing to delete anyway.